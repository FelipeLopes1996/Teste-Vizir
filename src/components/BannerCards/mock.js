export const mockInfoCards = [
  {
    title: "Plano Fale + 30",
    content: `Fale de graça até os 30 minutos iniciais de sua ligação, e só paga os
                     minutos excedentes. Os minutos excedentes tem um acrescimo de 
                     10% sobre a tarifa normal do minuto`,
  },
  {
    title: "Plano Fale + 60",
    content: `Fale de graça até os 60 minutos iniciais de sua ligação, e só paga os
      minutos excedentes. Os minutos excedentes tem um acrescimo de 
      10% sobre a tarifa normal do minuto`,
  },
  {
    title: "Plano Fale + 120",
    content: `Fale de graça até os 120 minutos iniciais de sua ligação, e só paga os
      minutos excedentes. Os minutos excedentes tem um acrescimo de 
      10% sobre a tarifa normal do minuto`,
  },
];
