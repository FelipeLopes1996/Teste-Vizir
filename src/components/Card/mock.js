export const mockInfoCard = [
  {
    title: "Plano Fale + 30",
    content: `text 1`,
  },
  {
    title: "Plano Fale + 60",
    content: `text 2`,
  },
  {
    title: "Plano Fale + 120",
    content: `text 3`,
  },
];
