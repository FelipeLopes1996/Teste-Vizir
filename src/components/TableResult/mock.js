export const ResultMock = [
  {
    origin: "origin Result",
    destinyChoice: "destinyChoice Result",
    minutes: "minutes Result",
    selectPlan: "selectPlan Result",
    validResultWithPlan: "validResultWithPlan Result",
    totalResult: "totalResult Result",
  },
  {
    origin: "origin Result",
    destinyChoice: "destinyChoice Result",
    minutes: "minutes Result",
    selectPlan: "selectPlan Result",
    validResultWithPlan: "validResultWithPlan Result",
    totalResult: "totalResult Result",
  },
];
