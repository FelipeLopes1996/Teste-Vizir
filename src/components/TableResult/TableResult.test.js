import React from "react";
import { render, screen } from "@testing-library/react";
import { TableResult } from ".";
import { ResultMock } from "./mock";

describe("<TableResult />", () => {
  it("should render title", () => {
    render(<TableResult />);
    const element = screen.getByText(/Resultado do cÃ¡lculo/i);
    expect(element).toBeInTheDocument();
  });
  it("should render Th values", () => {
    render(<TableResult />);
    const thOrigin = screen.getByText(/Origem/i);
    expect(thOrigin).toBeInTheDocument();
    const thDestiny = screen.getByText(/Destino/i);
    expect(thDestiny).toBeInTheDocument();
    const thTime = screen.getByText(/Tempo/i);
    expect(thTime).toBeInTheDocument();
    const thPlan = screen.getByText(/Plano FaleMais/i);
    expect(thPlan).toBeInTheDocument();
    const thWithPlan = screen.getByText(/Com FaleMais/i);
    expect(thWithPlan).toBeInTheDocument();
    const thNoPlan = screen.getByText(/Sem FaleMais/i);
    expect(thNoPlan).toBeInTheDocument();
  });

  it("should render MapResult values", () => {
    render(<TableResult mapResult={ResultMock} />);
    const originResult = screen.getAllByText(/origin Result/i);
    expect(originResult).toHaveLength(2);
    const destinyChoiceResult = screen.getAllByText(/destinyChoice Result/i);
    expect(destinyChoiceResult).toHaveLength(2);
    const minutesResult = screen.getAllByText(/minutes Result/i);
    expect(minutesResult).toHaveLength(2);
    const selectPlansResult = screen.getAllByText(/selectPlan Result/i);
    expect(selectPlansResult).toHaveLength(2);
    const validResultWithPlanResult = screen.getAllByText(
      /validResultWithPlan Result/i
    );
    expect(validResultWithPlanResult).toHaveLength(2);
    const totalResult = screen.getAllByText(/totalResult Result/i);
    expect(totalResult).toHaveLength(2);
  });
  it("should render snapshot", () => {
    const { container } = render(<TableResult />);
    expect(container).toMatchSnapshot();
  });
});
