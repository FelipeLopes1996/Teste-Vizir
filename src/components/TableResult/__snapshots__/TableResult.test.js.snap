// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableResult /> should render snapshot 1`] = `
<div>
  <div
    class="sc-bdvvtL dAepRo"
  >
    <h1>
      Resultado do c√°lculo
    </h1>
    <div>
      <table
        class="sc-gsDKAQ gCUYtx"
      >
        <thead>
          <tr
            class="sc-dkPtRN cIDnkU"
          >
            <th
              class="sc-hKwDye bDETGm"
            >
              Origem
            </th>
            <th
              class="sc-hKwDye bDETGm"
            >
              Destino
            </th>
            <th
              class="sc-hKwDye bDETGm"
            >
              Tempo
            </th>
            <th
              class="sc-hKwDye bDETGm"
            >
              Plano FaleMais
            </th>
            <th
              class="sc-hKwDye bDETGm"
            >
              Com FaleMais
            </th>
            <th
              class="sc-hKwDye bDETGm"
            >
              Sem FaleMais
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
